platform/security/filter/security-filter-csrf/src/main/java/org/codice/ddf/security/filter/csrf/CsrfFilter.java

```diff
@@ -82,29 +104,32 @@ public void doFilter(ServletRequest request, ServletResponse response, FilterCha
     // Begin CSRF checks if request is accessing a Cross-Site protected context
     if (protectedContexts.stream().anyMatch(targetContextPath::startsWith)) {
 
-      String targetOrigin = httpRequest.getRequestURL().toString();
       String sourceOrigin = httpRequest.getHeader(ORIGIN_HEADER);
       String sourceReferer = httpRequest.getHeader(REFERER_HEADER);
       String csrfHeader = httpRequest.getHeader(CSRF_HEADER);
 
       // Reject if no origin or referer header is present on the request
       if (sourceOrigin == null && sourceReferer == null) {
         respondForbidden(
-            httpResponse, "Incoming request did not have an Origin or Referer header.");
+            httpResponse,
+            "Cross-site check failure: Incoming request did not have an Origin or Referer header.");
         return;
       }
 
-      // Reject if neither the Origin or Referer header match the target origin
-      if (!isSameOrigin(sourceOrigin, targetOrigin) && !isSameOrigin(sourceReferer, targetOrigin)) {
+      // Reject if neither the Origin or Referer header are a trusted authority
+      if (!isTrustedAuthority(sourceOrigin) && !isTrustedAuthority(sourceReferer)) {
         respondForbidden(
-            httpResponse, "Origin or Referer header of request did not match target origin.");
+            httpResponse,
+            "Cross-site check failure: Neither the Origin or Referer header matched a system internal or external name.");
```

Comment:
"nor" instead of "or"

Reply:
All neither-ors have been updated to neither-nors